src/MonitoringFunctions/DataService/Kusto/KustoTable.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Kusto.Data;
+using Kusto.Data.Common;
+using Kusto.Ingest;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions.DataService.Kusto
+{
+    internal sealed class KustoTable<T> where T : IKustoTableRow
+    {
+        private KustoConnectionStringBuilder _connectionStringBuilder;
+
+        private KustoQueuedIngestionProperties _ingestionProperties;
+
+        public KustoTable(KustoConnectionStringBuilder connectionStringBuilder, string databaseName, string tableName, IJsonColumnMappingResolver columnMappingResolver)
+            : this(connectionStringBuilder, databaseName, tableName, columnMappingResolver.GetColumnMappings<T>())
+        {
+
+        }
+
+        public KustoTable(KustoConnectionStringBuilder connectionStringBuilder, string databaseName, string tableName, IEnumerable<ColumnMapping> columnMappings)
+        {
+            _connectionStringBuilder = connectionStringBuilder;
+
+            _ingestionProperties = new KustoQueuedIngestionProperties(databaseName, tableName)
+            {
+                ReportLevel = IngestionReportLevel.FailuresOnly,
+                ReportMethod = IngestionReportMethod.Queue,
+                IngestionMapping = new IngestionMapping()
+                {
+                    IngestionMappingKind = global::Kusto.Data.Ingestion.IngestionMappingKind.Json,
+                    IngestionMappings = columnMappings
+                },
+                Format = DataSourceFormat.json
+            };
+        }
+
+        public async Task InsertRow(T row, CancellationToken cancellationToken = default)
+        {
+            using IKustoQueuedIngestClient ingestClient = KustoIngestFactory.CreateQueuedIngestClient(_connectionStringBuilder);
+
+            string serializedData = JsonConvert.SerializeObject(row);
+            byte[] serializedBytes = Encoding.UTF8.GetBytes(serializedData);
+
+            using MemoryStream dataStream = new MemoryStream(serializedBytes);
+
+            // IKustoQueuedIngestClient doesn't support cancellation at the moment. Update the line below if it does in the future.
+            await ingestClient.IngestFromStreamAsync(dataStream, _ingestionProperties, leaveOpen: true);
```

Comment:
minor:  in cases like this, where an instance is disposable, created with a  ```using``` and then passed to a method, sometimes results in an unexpected disposed instance in the called method's stack.
**Question** have you tested this behavior?  My solution tosuch cases has been removing the using.

Reply:
Yes, this has been running on Azure for a couple days with no issues.
I think you can see that behavior if you forget to await the call, but otherwise, I would consider that a runtime bug. I would be very interested to see this, if you know of a sample that we can reproduce this with.